### ISSUE TITLE
Can't reach kubernetes using the deployed public ip 

### ISSUE THREAD

**Issue by waleedhammam** [threefoldtech/zos#1202]
After investigation, the vdc has gone out of grace period
![Screenshot from 2021-02-28 10-52-16](https://user-images.githubusercontent.com/10658229/109412864-19916d00-79b3-11eb-8dfc-e92e8d59ffac.png)

Yet I can't reach the public IP
```
(js-sdk-vpmH-Z_5-py3.8) root@vdc-testnet:~/js-sdk# ping 185.69.167.185
PING 185.69.167.185 (185.69.167.185) 56(84) bytes of data.
^C
--- 185.69.167.185 ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 1017ms
```

Kubernetes workloads: https://explorer.testnet.grid.tf/api/v1/reservations/workloads/5736
ip workload: https://explorer.testnet.grid.tf/api/v1/reservations/workloads/5735
Pool already has ipv4 units: https://explorer.testnet.grid.tf/api/v1/reservations/pools/2141

ip routes
```
default via 185.69.167.1 dev eth1 
10.42.0.0/24 dev cni0 proto kernel scope link src 10.42.0.1 
10.42.1.0/24 via 10.42.1.0 dev flannel.1 onlink 
10.42.2.0/24 via 10.42.2.0 dev flannel.1 onlink 
10.200.0.0/16 via 10.200.2.1 dev eth0 
10.200.2.0/24 dev eth0 proto kernel scope link src 10.200.2.2 
100.64.0.0/10 via 10.200.2.1 dev eth0 
127.0.0.0/8 dev lo scope host 
185.69.167.0/24 dev eth1 proto kernel scope link src 185.69.167.185 
```

iptables
```
default via 185.69.167.1 dev eth1 
10.42.0.0/24 dev cni0 proto kernel scope link src 10.42.0.1 
10.42.1.0/24 via 10.42.1.0 dev flannel.1 onlink 
10.42.2.0/24 via 10.42.2.0 dev flannel.1 onlink 
10.200.0.0/16 via 10.200.2.1 dev eth0 
10.200.2.0/24 dev eth0 proto kernel scope link src 10.200.2.2 
100.64.0.0/10 via 10.200.2.1 dev eth0 
127.0.0.0/8 dev lo scope host 
185.69.167.0/24 dev eth1 proto kernel scope link src 185.69.167.185 
k3os-16701 [~]$ sudo iptables -S
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
-N KUBE-EXTERNAL-SERVICES
-N KUBE-FIREWALL
-N KUBE-FORWARD
-N KUBE-KUBELET-CANARY
-N KUBE-PROXY-CANARY
-N KUBE-SERVICES
-A INPUT -m conntrack --ctstate NEW -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
-A INPUT -m conntrack --ctstate NEW -m comment --comment "kubernetes externally-visible service portals" -j KUBE-EXTERNAL-SERVICES
-A INPUT -j KUBE-FIREWALL
-A FORWARD -m comment --comment "kubernetes forwarding rules" -j KUBE-FORWARD
-A FORWARD -m conntrack --ctstate NEW -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
-A FORWARD -s 10.42.0.0/16 -j ACCEPT
-A FORWARD -d 10.42.0.0/16 -j ACCEPT
-A OUTPUT -m conntrack --ctstate NEW -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
-A OUTPUT -j KUBE-FIREWALL
-A KUBE-EXTERNAL-SERVICES -p tcp -m comment --comment "kube-system/traefik:websecure has no endpoints" -m addrtype --dst-type LOCAL -m tcp --dport 30406 -j REJECT --reject-with icmp-port-unreachable
-A KUBE-EXTERNAL-SERVICES -p tcp -m comment --comment "kube-system/traefik:web has no endpoints" -m addrtype --dst-type LOCAL -m tcp --dport 32487 -j REJECT --reject-with icmp-port-unreachable
-A KUBE-FIREWALL -m comment --comment "kubernetes firewall for dropping marked packets" -m mark --mark 0x8000/0x8000 -j DROP
-A KUBE-FIREWALL ! -s 127.0.0.0/8 -d 127.0.0.0/8 -m comment --comment "block incoming localnet connections" -m conntrack ! --ctstate RELATED,ESTABLISHED,DNAT -j DROP
-A KUBE-FORWARD -m conntrack --ctstate INVALID -j DROP
-A KUBE-FORWARD -m comment --comment "kubernetes forwarding rules" -m mark --mark 0x4000/0x4000 -j ACCEPT
-A KUBE-FORWARD -m comment --comment "kubernetes forwarding conntrack pod source rule" -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A KUBE-FORWARD -m comment --comment "kubernetes forwarding conntrack pod destination rule" -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A KUBE-SERVICES -d 10.43.155.32/32 -p tcp -m comment --comment "kube-system/traefik:websecure has no endpoints" -m tcp --dport 443 -j REJECT --reject-with icmp-port-unreachable
-A KUBE-SERVICES -d 10.200.2.2/32 -p tcp -m comment --comment "kube-system/traefik:websecure has no endpoints" -m tcp --dport 443 -j REJECT --reject-with icmp-port-unreachable
-A KUBE-SERVICES -d 10.43.155.32/32 -p tcp -m comment --comment "kube-system/traefik:web has no endpoints" -m tcp --dport 80 -j REJECT --reject-with icmp-port-unreachable
-A KUBE-SERVICES -d 10.200.2.2/32 -p tcp -m comment --comment "kube-system/traefik:web has no endpoints" -m tcp --dport 80 -j REJECT --reject-with icmp-port-unreachable
```

**Comment by LeeSmet** [comment]
The VM is sending out an ARP request for the router every second (and getting a reply). I'd need access to the VM to do some further investigations

**Comment by xmonader** [comment]
@waleedhammam  please arrange with Lee to have access to the VM

**Comment by waleedhammam** [comment]
> @waleedhammam please arrange with Lee to have access to the VM

already done (y)

**Comment by LeeSmet** [comment]
Packet dumping inside the VM show that the ARP reply is not received (or is already dropped before it reaches the packet filter). Arptables are empty however

**Comment by DylanVerstraete** [comment]
@LeeSmet @waleedhammam status update?

**Comment by waleedhammam** [comment]
It happens again today
```
JS-NG> j.sals.vdc.vdc_samar_samar.kubernetes                                                                                                                                                                  
[KubernetesNode(
  role=<KubernetesRole.MASTER: 'master'>,
  public_ip='185.69.167.126',
  _size=15,
  ip_address='10.200.2.2',
  wid=34160,
  node_id='BpTAry1Na2s1J8RAHNyDsbvaBSM3FjR4gMXEKga3UPbs',
  pool_id=8347
), KubernetesNode(
  role=<KubernetesRole.WORKER: 'worker'>,
  public_ip='::/128',
  _size=16,
  ip_address='10.200.3.2',
  wid=34167,
  node_id='48YmkyvfoXnXqEojpJieMfFPnCv2enEuqEJcmhvkcdAk',
  pool_id=8347
)]
➜  js-sdk git:(development_vdc_backup_restore) ✗ ping 185.69.167.126           
PING 185.69.167.126 (185.69.167.126) 56(84) bytes of data.
^C
--- 185.69.167.126 ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 1009ms
```

I'll send @LeeSmet  the configs

**Comment by LeeSmet** [comment]
Some initial analysis of the syscalls performed by the firecracker VM seem to suggest that no `read` syscalls are done on one of the 2 tap fd's. When checking the fds, there are indeed 2 tap's (both fd 18 and 25 link into `/dev/net/tun`). After letting a strace run for a few seconds, we see both read and write syscalls on fd 18, but only writes on 25 (write would be outgoing network traffic as the process writes the packet into the tap fd, read is supposed to be incoming as the vm process reads the packet coming into the vm from the tap).

More debugging as to what causes this will be needed, though at this point we are likely dealing with a firecracker issue (as the reply packets are visible on the tap when running tcpdump)

**Comment by MohamedEssamAdawy** [comment]
It happens again with my vdc on devnet with the following configurations
```
JS-NG> essamvdc.kubernetes                                                                                            
[KubernetesNode(
  role=<KubernetesRole.MASTER: 'master'>,
  public_ip='185.69.167.134',
  _size=15,
  ip_address='10.200.2.2',
  wid=34245,
  node_id='2anfZwrzskXiUHPLTqH1veJAia8G6rW2eFLy5YFMa1MP',
  pool_id=8376
), KubernetesNode(
  role=<KubernetesRole.WORKER: 'worker'>,
  public_ip='::/128',
  _size=17,
  ip_address='10.200.3.2',
  wid=34258,
  node_id='3NAkUYqm5iPRmNAnmLfjwdreqDssvsebj4uPUt9BxFPm',
  pool_id=8376
), KubernetesNode(
  role=<KubernetesRole.WORKER: 'worker'>,
  public_ip='::/128',
  _size=17,
  ip_address='10.200.4.2',
  wid=34279,
  node_id='BpTAry1Na2s1J8RAHNyDsbvaBSM3FjR4gMXEKga3UPbs',
  pool_id=8376
), KubernetesNode(
  role=<KubernetesRole.WORKER: 'worker'>,
  public_ip='::/128',
  _size=17,
  ip_address='10.200.5.2',
  wid=34292,
  node_id='BBmMHCw392RLiByVEUT2ac6y6qJpbW9xj8TRdgk5rMGN',
  pool_id=8376
)]

```

**Comment by LeeSmet** [comment]
Probably fixed by #1210
